### ISSUE TITLE
C function receives invalid n from nquad

### ISSUE THREAD

**Issue by iljah** [scipy/scipy#7129]
I compile this c file:

```
#include "stdio.h"
double f(int n, double args[]) {
	printf("%i\n", n);
	return 0;
}
```
with ```gcc -fPIC -shared func.c -o func.so``` and run this python file 4 times:
```
#!/usr/bin/env python3
import ctypes
from scipy.integrate import nquad
lib = ctypes.CDLL('func.so')
func = lib.f
func.restype = ctypes.c_double
func.argtypes = (ctypes.c_int, ctypes.c_double)
print(nquad(func, [[0, 1]]))
```
and it prints:
```
$ ./main.py
32765
...
32765
(0.0, 0)
$ ./main.py 
32765
...
32765
(0.0, 0)
$ ./main.py
32764
...
32764
(0.0, 0)
$ ./main.py 
32766
...
32766
(0.0, 0)
```
what could be wrong? Apparently n should be 1 here. I'm using:
```
Linux mun3 4.9.12-200.fc25.x86_64 #1 SMP Thu Feb 23 19:31:49 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
gcc (GCC) 6.3.1 20161221 (Red Hat 6.3.1-1)
Python 3.5.2
scipy 0.18.0
```

**Comment by pv** [comment]
This is probably fixed in 0.19.0rc2. However, `n` is actually not the 
number of dimensions, but the length of the `args` array, which contains 
the coordinates, followed by any additional numbers passed in via the 
args= argument.

**Comment by ev-br** [comment]
> This is probably fixed in 0.19.0rc2.

And so it is:

```
$ python func.py 
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
1
(0.0, 0)
br@duneyrr:~/sweethome/temp$ python --version
Python 2.7.11
br@duneyrr:~/sweethome/temp$ python -c'import scipy; print scipy.__version__'
0.19.0.dev0+997f371
```

with `func.c` and `func.py` from the OP.
